### PR TITLE
NEW MFA module now has a logger and test state to clear user defined handlers in unit testing

### LINKED ISSUE #79: Add a generic MFA logger for authenticators to use
`LoggerInterface::class` - maybe the core one, maybe a custom one for MFA specifically.

Consideration also for how this will work with audit logging hooks which will a similar kind of thing.

Add it everywhere that exceptions are caught

## Pull requests

- [x] #101
- [x] https://github.com/silverstripe/silverstripe-totp-authenticator/pull/9

### PR DESCRIPTION
**Pull request by robbieaverill**
Resolves #79

### CODE REVIEW

File: src/Dev/State/LoggerTestState.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        $logger = Injector::inst()->get(LoggerInterface::class . '.mfa');
+        $logger->setHandlers([new NullHandler()]);
+    }
+
+    public function tearDown(SapphireTest $test)
+    {
+        // noop
+    }
+
+    public function setUpOnce($class)
```
Comment (NightJar) [inline src/Dev/State/LoggerTestState.php]: What is the point of this method (and `tearDownOnce`)?
As far as I know these interfaces are defined neither in PHPUnit nor SapphireTest

Comment (robbieaverill) [reply src/Dev/State/LoggerTestState.php]: They're methods in `TestState`

**Comment by robbieaverill** [comment]
The TestState is there to replace any user configured log handlers with a NullHandler during unit test runs, so unit tests don't pollute your MFA log handlers with their output. I can improve the TestState class docblock if what's there is unclear

File: src/Dev/State/LoggerTestState.php (hunk 1 above)
Comment (NightJar) [reply src/Dev/State/LoggerTestState.php]: Ohh, I've completely misread the purpose of this class 😂

**Pull request merged**